### ISSUE TITLE
We Should be Able to Pass `-docker.enabled` from the Command Line

### ISSUE THREAD

**Issue by jeffquinn-msk** [nextflow-io/nextflow#4363]
From the command line interface, it seems we can set configuration values in the `process` block https://github.com/nextflow-io/nextflow/blob/5e3f2c0aac33c974c427b1d252acd6f17cbc3ea1/modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy#L123

And in the `executor` block https://github.com/nextflow-io/nextflow/blob/5e3f2c0aac33c974c427b1d252acd6f17cbc3ea1/modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy#L132

But why can't we set configuration values in the `docker`, `singularity`, etc blocks? Unless I'm misunderstanding the set of configuration values that can be specified from the `.config` file does not line up with the set of configuration values that can be specified from the CLI.

**Comment by bentsherman** [comment]
Indeed, the support for config options on the command line is ad-hoc, probably added over time as needed. Aside from `process` and `executor`, there are a bunch of `-with-*` options to enable things like docker, singularity, wave, etc. So `-with-docker` is the same as `docker.enabled = true` in the config file.

We are developing a new CLI (#3595) where we will be able to redesign these options. Feel free to give feedback there if you have any suggestions.
